configuration/outbound.md

```diff
@@ -0,0 +1,201 @@
+
+Example:
+
+```json
+{
+  "destination": {
+    "base_url": "https://example.com",
+    "auth": {
+      "type": "basic",
+      "username": "admin",
+      "password_key": "example.com"
+    },
+    "path": "/api/v1/referral"
+  }
+}
+```
+
+Here `password_key` is a key used to find the password in CouchDB's node-based configuration. See Credentials section below.
+
+If you don't provide an authentication parameter then the request will be sent without authentication.
+
+### mapping
+
+A complex property that declares how the payload to be sent to the `destination` should be created.
+
+Each key is a JSON path to a location in the payload, and each value is either:
+ - a string JSON path to the location of the source data in the report being processed, where the value is required to exist
+ - an object with either a `path` property that represents a JSON path as above, or an `expr` property which is an expression similar to `relevant_to` to determine the resulting value. If you wish for the value to be optional (that is to say it's OK if `path` or `expr` evaluate to `undefined`) you may also set `optional` to `true`
+
+#### A mapping example
+
+Given the following example (incomplete) report:
+```json
+{
+  "id": "abc-1234",
+  "type": "data_record",
+  "form": "referral",
+  "reported_date": 1555069530966,
+  "fields": {
+    "patient_id": "12345",
+    "patient_temp_deg_F": 100,
+    "danger_signs": ["V", "BL", "D", "IG"]
+  },
+  "contact": {
+    "_id": "def-5678",
+    "name": "Jane CHW",
+    "parent": {
+      "_id": "hij-9012",
+      "name": "..."
+    }
+  }
+}
+```
+
+The following mapping configuration:
+```json
+{
+  "mapping": {
+    "patient_id": "doc.fields.patient_id",
+    "chw_id": "doc.contact._id",
+    "event.temp_c": {
+      "expr": "((doc.fields.patient_temp_deg_F - 32) * (5.0/9.0)).toFixed(2)"
+    },
+    "event.in_danger": {
+      "expr": "doc.fields.danger_signs.length >= 3 ? true : undefined",
+      "optional": true
+    },
+    "event.additional_notes": {
+      "path": "doc.fields.notes",
+      "optional": true
+    }
+  }
+}
+```
+
+Would create the following JSON payload to send:
+```json
+{
+  "patient_id": "12345",
+  "chw_id": "def-5678",
+  "event": {
+    "temp_c": 37.78,
+    "in_danger": true
+  }
+}
+```
+
+This example makes a few points:
+ - The report that is being used to generate the outbound push is referenced as `doc` in both `path` and `expr` properties
+ - To define a property that is itself an object, you can make the mapping key a JSON path.
+ - If you define a property as `optional`, it won't exist at all in the payload if the resulting value is `undefined`, either because that is result of executing the `expr`, or the `path` doesn't exist. Note that if the `event.in_danger` expression was instead `doc.fields.danger_signs.length >= 3` the property `in_danger` would always exist and would either be `true` or `false`
+
+Other notes:
```

Comment:
Should this be a new section rather than including these notes under the "example" section?